src/d/llvm/expression.d

```diff
@@ -704,7 +710,45 @@ struct ExpressionGen {
 				
 				auto result = buildCall(declare(pass.object.getClassDowncast()), args[]);
 				return LLVMBuildBitCast(builder, result, type, "");
-			
+
+			case IntToFloat:
+				const toType = e.type.builtin();
+				//Should be correct long before reaching here
+				assert(isFloat(toType));
+				const srcType = e.expr.type.builtin();
+				import std.stdio;
+				//D considers char et al to be an integral type, we do not (yet)
+				const BuiltinType underlyingType = isChar(srcType) ? integralOfChar(srcType) : srcType;
+				if (isSigned(underlyingType))
```

Comment:
Considering there is a distinction between SPad and UPad, it would make sense to me to have two types of casts here too. You wouldn't have to do all the type dance above, and bonus point, it won't break on aliased types.